bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/obs/SPObsCache.java

```diff
@@ -49,6 +50,16 @@ public static void setTargetCalculator(ISPObservation obs, Option<TargetCalculat
         setObsCache(obs, cache);
     }
 
+    public static Option<Instrument> getInstrument(ISPObservation obs) {
+        return ImOption.apply(getObsCache(obs)).flatMap(c -> c.getInstrument());
+    }
+
+    public static void setInstrument(ISPObservation obs, Option<Instrument> instrument) {
```

Comment:
Can you remind me the circumstances under which the obs cache is invalidated?

Reply:
[Whenever anything inside of an observation tree changes](https://github.com/gemini-hlsw/ocs/blob/develop/bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/obs/SPObsCache.java#L317-L325).